exporter/exporterhelper/internal/queue/persistent_queue_test.go

```diff
@@ -216,16 +219,38 @@ func (m *fakeStorageClientWithErrors) Reset() {
 	m.nextErrorIndex = 0
 }
 
+type fakeReferenceCounter struct {
+	mu  sync.Mutex
+	ref int64
+}
+
+func (f *fakeReferenceCounter) Ref(int64) {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	f.ref++
+}
+
+func (f *fakeReferenceCounter) Unref(int64) {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	f.ref--
+	if f.ref < 0 {
+		panic("this should never happen")
```

Comment:
I'm afraid panic is too much given that this is publicly exposed as `Unref<signal>`.

Reply:
If you are worried, while we have the pooling gate disabled we can disable the panic, does that help? After polling is enabled getting to this point will be undefined behavior since the pointer is freed

Reply:
I guessed that, and for that logic I answered. It is not correct, but as said this is a very rare and unsafe API (indeed), so I am not sure how else I can protect (also for the moment in xpdata so we can still iterate). Right now:

```
m := pmetric.NewMetrics()
pref.UnrefMetrics(m)
m.ResourceMetrics() -> will panic here since the orig is deallocated/nil.
pref.UnrefMetrics(m)
``

Reply:
Ok sounds good